pkg/policy/repository.go

```diff
@@ -167,19 +167,27 @@ func NewPolicyRepository(
 	certManager certificatemanager.CertificateManager,
 	secretManager certificatemanager.SecretManager,
 ) *Repository {
-	repoChangeQueue := eventqueue.NewEventQueueBuffered("repository-change-queue", option.Config.PolicyQueueSize)
-	ruleReactionQueue := eventqueue.NewEventQueueBuffered("repository-reaction-queue", option.Config.PolicyQueueSize)
-	repoChangeQueue.Run()
-	ruleReactionQueue.Run()
-	selectorCache := NewSelectorCache(idAllocator, idCache)
+	repo := NewStoppedPolicyRepository(idAllocator, idCache, certManager, secretManager)
+	repo.Start()
+	return repo
+}
 
+// NewStoppedPolicyRepository creates a new policy repository.
+//
+// Qeues must be allocated via [Repository.Start]. The function serves to
+// satisfy hive invariants.
+func NewStoppedPolicyRepository(
```

Comment:
What does "Stopped" mean? I found out what it means by reading further along in the code, but it would be good to document it in this function's godoc.

Reply:
```
$ git grep NewPolicyReposit | grep -v test.go
daemon/cmd/daemon.go:           // Must be done before calling policy.NewPolicyRepository() below.
pkg/policy/repository.go:// NewPolicyRepository creates a new policy repository.
pkg/policy/repository.go:func NewPolicyRepository(
$ git grep NewPolicyReposit | grep test.go | wc -l
68
```

It looks like the main NewPolicyRepository() is now just a helper for test code. It's a bit of a code smell to me to call this `NewStoppedPolicyRepository()` rather than just having a `New...()` + `Start()`, but I appreciate that all those other test code references are a bit of a pain to fix up if they're all relying on this code as-is to initialize+start the policy repo. I'm OK to defer this for now.

Reply:
I've updated the function doc. I think Jussi's preference was to pass a `hive.Lifecycle` into the constructor, but that causes a big ripple effect, NewPolicyRepository is used a lot. So I did this botch to keep the changes small(er).